src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Frameworks/ITargetFrameworkProvider.cs

```diff
@@ -3,10 +3,10 @@
 using System.Collections.Generic;
 using Microsoft.VisualStudio.Composition;
 
-namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.CrossTarget
+namespace Microsoft.VisualStudio.ProjectSystem
 {
-    [ProjectSystemContract(ProjectSystemContractScope2.Global, ProjectSystemContractProvider.Private, Cardinality = ImportCardinality.ExactlyOne)]
-    internal interface ITargetFrameworkProvider
+    [ProjectSystemContract(ProjectSystemContractScope2.Global, ProjectSystemContractProvider.Host, Cardinality = ImportCardinality.ExactlyOne)]
+    public interface ITargetFrameworkProvider
```

Comment:
Ditto (public).

Reply:
Agreed, we don't want this public.

Reply:
The contract between the core layer and VS layer must follow one of the two conditions below:
 1, the contract is public;
 2, the contract is optional, and not essential to use the core layer alone.

Because this is required MEF component to make the core layer to work, so it must be public, or anyone cannot use the core layer alone.  Another choice is to move the implementation inside the core layer (without depending on any VS function).  Of course, we can also copy the implementation from NuGet VS to your codebase (which is still a shim layer to wire to the NuGet framework assembly.)  I tried this, but it would require to load the NuGet assembly, which is not listed in the VS searching path at this moment, so it requires additional product change (+ more duplicated code.)

Of course, that will be your call to decide which router we want to go...

Reply:
I see. I don't see a way around this @davkean. It's only adding two new types to the API surface: `ITargetFrameworkProvider` and `ITargetFramework`.

Reply:
Let me think about this; this is code I had planned on throwing away.

Reply:
I think we can keep them as internal for now, if you need throw it away in the future. The separation of two layers are more for long term future, instead of being required immediately.

Reply:
Let me know what is your decision. I can revert it to keep them internal, and we can resolve the problem for now, and move on for other changes.

Reply:
Yep, let's do that and file a bug against me to fix this "internal" relationship.

Reply:
Reverted accessibility in fa14ef8875b714e9fce8a482dc79f3ecaed7e1a8 and opened #4886 to track removing those interfaces.